src/app/prices/get-current-price.ts

```diff
@@ -1,29 +1,53 @@
+import { SECS_PER_10_MINS } from "@config"
+
 import { CacheKeys } from "@domain/cache"
-import { toSeconds } from "@domain/primitives"
+import { PriceNotAvailableError } from "@domain/price"
+import { checkedToDisplayCurrency } from "@domain/fiat"
+
 import { PriceService } from "@services/price"
 import { LocalCacheService } from "@services/cache"
-import { PriceNotAvailableError } from "@domain/price"
 
-export const getCurrentPrice = async (): Promise<
-  DisplayCurrencyPerSat | PriceServiceError
-> => {
-  const realtimePrice = await PriceService().getRealTimePrice()
-  if (realtimePrice instanceof Error) return getCachedPrice()
-  // keep price in cache for 10 mins in case the price pod is not online
-  await LocalCacheService().set<DisplayCurrencyPerSat>({
+export const getCurrentPrice = async ({
+  currency,
+}: GetCurrentPriceArgs): Promise<DisplayCurrencyPerSat | ApplicationError> => {
+  const checkedDisplayCurrency = checkedToDisplayCurrency(currency)
+  if (checkedDisplayCurrency instanceof Error) return checkedDisplayCurrency
+
+  const realtimePrice = await PriceService().getRealTimePrice({
+    currency: checkedDisplayCurrency,
+  })
+  if (realtimePrice instanceof Error)
+    return getCachedPrice({ currency: checkedDisplayCurrency })
+
+  let cachedPrices = await getCachedPrices()
+  cachedPrices = cachedPrices instanceof Error ? {} : cachedPrices
+
+  cachedPrices[currency] = realtimePrice
+
+  // keep prices in cache for 10 mins in case the price pod is not online
+  await LocalCacheService().set<DisplayCurrencyPrices>({
     key: CacheKeys.CurrentPrice,
-    value: realtimePrice,
-    ttlSecs: toSeconds(600),
+    value: cachedPrices,
```

Comment:
well I had the same question but since price server calc is based on USD then I decided to use only 1 cache key for all currencies. the only problem could be a stale price in a currency different to USD but in general FX quotes are updated 1 or twice a day.